src/Mod/Sketcher/App/ExternalGeometryFacade.cpp

```diff
@@ -160,6 +161,31 @@ void ExternalGeometryFacade::copyId(const Part::Geometry* src, Part::Geometry* d
     gfdst->setId(gfsrc->getId());
 }
 
+void ExternalGeometryFacade::copyFlags(const Part::Geometry* src, Part::Geometry* dst)
+{
+    auto gfsrc = ExternalGeometryFacade::getFacade(src);
+    auto gfdst = ExternalGeometryFacade::getFacade(dst);
+    gfdst->setFlags(gfsrc->getFlags());
+}
+
+void ExternalGeometryFacade::setRef(const std::string& ref)
+{
+    if (ref.size() && getId() < 0) {

```

Comment:
I'm surprised the CI's linter isn't complaining here about the int-to-bool conversion -- mine suggests using `!empty()`